### PR TITLE
MAILET-138 SenderIsLocal should be covered with tests

### PR DESCRIPTION
**Pull request by Laura-Royet**

### CODE REVIEW

File: mailet/standard/src/test/java/org/apache/james/transport/matchers/SenderIsLocalTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    @Test
+    public void shouldReturnRecipientsEmailsAdressWhenSenderIsLocal() throws MessagingException  {
+        Mail mail = FakeMail.builder()
+            .sender(ANY_AT_JAMES)
+            .recipient(ANY_AT_JAMES2)
+            .build();
+
+        assertThat(matcher.match(mail)).containsExactly(ANY_AT_JAMES2);
```
Comment (aduprat) [inline mailet/standard/src/test/java/org/apache/james/transport/matchers/SenderIsLocalTest.java]: extract tested method response

File: mailet/standard/src/test/java/org/apache/james/transport/matchers/SenderIsLocalTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    @Test
+    public void shouldReturnNullWhenSenderIsNotLocal() throws MessagingException {
+        Mail mail = FakeMail.builder()
+            .sender(ANY_AT_JAMES2)
+            .recipient(ANY_AT_JAMES)
+            .build();
+
+        assertThat(matcher.match(mail)).isNull();
```
Comment (aduprat) [inline mailet/standard/src/test/java/org/apache/james/transport/matchers/SenderIsLocalTest.java]: same here

File: mailet/standard/src/test/java/org/apache/james/transport/matchers/SenderIsLocalTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        when(mailContext.isLocalEmail(ANY_AT_JAMES)).thenReturn(true);
+        when(mailContext.isLocalEmail(ANY_AT_JAMES2)).thenReturn(false);
+        
+        matcher = new SenderIsLocal();
+        FakeMatcherConfig mci = new FakeMatcherConfig("SenderIsLocal", mailContext);
+        matcher.init(mci);
+    }
+
+    @Test
+    public void shouldReturnRecipientsEmailsAdressWhenSenderIsLocal() throws MessagingException  {
```
Comment (aduprat) [inline mailet/standard/src/test/java/org/apache/james/transport/matchers/SenderIsLocalTest.java]: What should do something?
You may add something before should: `matcherShouldReturn...`

**Comment by Laura-Royet** [comment]
test this please

**Pull request merged**